docs/DeveloperGuide.adoc

```diff
@@ -142,72 +145,79 @@ image::StorageClassDiagram.png[]
 
 The following sequence diagram shows how the undo operation works:
 
+.Interactions Inside the Logic Component for the `undo` Command
 image::UndoSequenceDiagram.png[]
 
 NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
-The `redo` command does the opposite -- it calls `Model#redoAddressBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the address book to that state.
+The `redo` command does the opposite -- it calls `VersionedBillboard#redo()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the billboard to that state.
 
 [NOTE]
-If the `currentStatePointer` is at index `addressBookStateList.size() - 1`, pointing to the latest address book state, then there are no undone address book states to restore. The `redo` command uses `Model#canRedoAddressBook()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
+If the `currentStatePointer` is at index `stateList.size() - 1`, pointing to the latest billboard state, then there are no undone billboard states to restore. The `redo` command uses `VersionedBillboard#isRedoable()` to check if this is the case. If so, it will return an error to the user rather than attempting to perform the redo.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+Step 5. The user then decides to execute the command `list`. Commands that do not modify the billboard, such as `list`, will usually not call `VersionedBillboard#commit()`, `VersionedBillboard#undo()` or `VersionedBillboard#redo()`. Thus, the `stateList` remains unchanged.
 
+.State of VersionedBillboard after "list" command
 image::UndoRedoState4.png[]
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+Step 6. The user executes `clear`, which calls `VersionedBillboard#commit()`. Since the `currentStatePointer` is not pointing at the end of the `stateList`, all billboard states after the `statePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/buy ...` command. This is the behavior that most modern desktop applications follow.
```

Comment:
If `clear` deletes all the states, then where does `current state` point to?

Reply:
The current state should point to bb3:Billboard, sorry for my typo. Just in case you are unclear about how it works, I will further explain it. If you already know how it works, please ignore below. If the user executes any other billboard editing command(add, delete etc), then all the state after the current state will be removed, then execute the command, and store it into the state list, current state pointer will point to that state.  And you will not able to redo cause the state pointer is pointing to the last state in the list.